service/src/main/java/uk/nhs/adaptors/gp2gp/transformJsonToXmlTool/TransformJsonToXml.java

```diff
@@ -0,0 +1,232 @@
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.DiagnosticReportMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.ObservationMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.SpecimenMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.parameters.EhrExtractTemplateParameters;
+import uk.nhs.adaptors.gp2gp.gpc.GetGpcStructuredTaskDefinition;
+
+public class TransformJsonToXml {
+
+    private static final String JSON_FILE_INPUT_PATH =
+        Paths.get("src/").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/input/";
+    private static final String XML_OUTPUT_PATH =
+        Paths.get("src/").toFile().getAbsoluteFile().getAbsolutePath() + "/../../transformJsonToXml/output/";
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private static final FhirParseService FHIR_PARSE_SERVICE = new FhirParseService();
+
+    public static void main(String[] args) throws Exception {
+
+        try {
+            var inputWrapper = getFiles();
+            for (int i = 0; i < inputWrapper.getJsonFileInputs().size(); i++) {
+                var jsonString = inputWrapper.getJsonFileInputs().get(i);
+                String xmlResult = mapJsonToXml(jsonString);
+                var fileName = inputWrapper.getJsonFileNames().get(i);
+                writeToFile(xmlResult, fileName);
+            }
+        } catch (Exception e) {
+            LOGGER.error("error: " + e.getMessage());
+        }
+        LOGGER.error("end");
+    }
+
+    private static InputWrapper getFiles() throws Exception {
+        File[] files = new File(JSON_FILE_INPUT_PATH).listFiles();
+        List<String> jsonStringInputs = new ArrayList<>();
+        List<String> fileNames = new ArrayList<>();
+
+        if (files == null || files.length == 0) {
+            throw new Exception("No json files found");
+        }
+
+        LOGGER.info("Processing " + files.length + " files from location: " + JSON_FILE_INPUT_PATH);
+
+        Arrays.stream(files)
+            .peek(file -> LOGGER.info("Parsing file: {}", file.getName()))
+            .filter(file -> FilenameUtils.getExtension(file.getName()).equalsIgnoreCase("json"))
+            .forEach(file -> {
+                String jsonAsString = null;
+                try {
+                    jsonAsString = readJsonFileAsString(JSON_FILE_INPUT_PATH + file.getName());
+                } catch (Exception e) {
+                    LOGGER.error("Could not read {}", file.getName());
+                }
+                jsonStringInputs.add(jsonAsString);
+                fileNames.add(file.getName());
+            });
+        return InputWrapper.builder().jsonFileInputs(jsonStringInputs).jsonFileNames(fileNames).build();
+    }
+
+    private static String mapJsonToXml(String jsonAsStringInput) throws Exception {
+        final Bundle bundle = new FhirParseService().parseResource(jsonAsStringInput, Bundle.class);
```

Comment:
try separate class initialisation here so class are loaded once at the start instead of every file